src/coreclr/src/jit/importer.cpp

```diff
@@ -9298,7 +9344,34 @@ GenTree* Compiler::impFixupStructReturnType(GenTree* op, CORINFO_CLASS_HANDLE re
     assert(!info.compIsVarArgs);
 
     // Is method returning a multi-reg struct?
-    if (varTypeIsStruct(info.compRetNativeType) && IsMultiRegReturnedType(retClsHnd))
+    if (varTypeIsStruct(info.compRetNativeType) && IsMultiRegReturnedType(retClsHnd, unmgdCallConv))
```

Comment:
Seems like we are inconsistent here about using `FEATURE_MULTIREG_RET`, should we add this to the XARCH case too?